src/EpiGimp/src/SceneCanvas.cpp

```diff
@@ -60,12 +61,37 @@ void SceneCanvas::onEvent(const sf::Event &event)
 void SceneCanvas::onTick(float deltaTime)
 {
     m_deltaTime = deltaTime;
+    std::string savePath{};
 
     if (sf::Keyboard::isKeyPressed(sf::Keyboard::Up)) updateView({0.f, -1.f});
     if (sf::Keyboard::isKeyPressed(sf::Keyboard::Left)) updateView({-1.f, 0.f});
     if (sf::Keyboard::isKeyPressed(sf::Keyboard::Down)) updateView({0.f, 1.f});
     if (sf::Keyboard::isKeyPressed(sf::Keyboard::Right)) updateView({1.f, 0.f});
 
+
+    /**
+     * save layers as an Image
+     */
+    savePath = menu.getSavePath();
+    if (!savePath.empty())
+    {
+        static const char *extList[4] = {"bmp", "png", "tga", "jpg"};
+        std::string extension = savePath.substr(savePath.find_last_of('.')+1);
+        
+        for (auto i = 0; i < 4; ++i)
+        {
+            if (extList[i] == extension)
+            {
+                if (squash().saveToFile(savePath) == false) {
```

Comment:
`squash()` is a heavy operation that only needs to be done once.

Reply:
Its done only once, since it only done when the extension matches. The loop is break anyway if the save succeed or fail